src/main/scala/com/fulcrumgenomics/vcf/MakeMixtureVcf.scala

```diff
@@ -0,0 +1,268 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.vcf
+
+import java.text.DecimalFormat
+import java.util
+import java.util.Collections
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.Io
+import com.fulcrumgenomics.vcf.MakeMixtureVcf.Sample
+import dagr.commons.util.LazyLogging
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.util.CollectionUtil
+import htsjdk.variant.variantcontext._
+import htsjdk.variant.variantcontext.writer.{Options, VariantContextWriter, VariantContextWriterBuilder}
+import htsjdk.variant.vcf._
+
+import scala.collection.JavaConversions.iterableAsScalaIterable
+import scala.collection.mutable
+
+
+object MakeMixtureVcf {
+  /** The format field for the allele fraction to be written to. */
+  val AlleleFractionField = "AF"
+  /** The filter that is placed on records where the full genotype cannot be computed. */
+  val UnknownGtFilter     = "unknown_gt"
+
+  val HeaderLines = Seq[VCFHeaderLine](
+    new VCFFilterHeaderLine(UnknownGtFilter, "GT and AF not known due to one or more input samples being no-called."),
+    new VCFFormatHeaderLine("AF", VCFHeaderLineCount.A, VCFHeaderLineType.Float, "Alt allele fraction.")
+  )
+
+  /** Case class to store a sample and a proportion in the mixture. */
+  private[vcf] case class Sample(name: String, proportion: Double = 0) {
+    override def toString: String = f"${name}@${proportion}%.4f"
```

Comment:
Validate that proportion is > 0?  Does it make sense to have sample with proportion 0 (or less than zero)?